app/controllers/DataRetrievals.scala

```diff
@@ -70,4 +76,106 @@ object DataRetrievals {
     }
   }
 
+  def cyaChargeGeneric[A](chargeDetailsPage: QuestionPage[A],
+                          srn: String)
+                         (block: (A, String) => Future[Result])
+                         (implicit request: DataRequest[AnyContent], reads: Reads[A]): Future[Result] = {
+    (
+      request.userAnswers.get(chargeDetailsPage),
+      request.userAnswers.get(SchemeNameQuery)
+    ) match {
+      case (Some(chargeDetails), Some(schemeName)) =>
+        block(chargeDetails, schemeName)
+      case _ =>
+        Future.successful(Redirect(controllers.routes.AFTSummaryController.onPageLoad(srn, None)))
+    }
+  }
+
+  def cyaChargeC(isSponsoringEmployerIndividualPage: QuestionPage[Boolean],
+                 sponsoringIndividualDetailsPage: QuestionPage[MemberDetails],
+                 sponsoringOrganisationDetailsPage: QuestionPage[SponsoringOrganisationDetails],
+                 sponsoringEmployerAddressPage: QuestionPage[SponsoringEmployerAddress],
+                 chargeDetailsPage: QuestionPage[ChargeCDetails],
+                 srn: String)
+                (block: (Boolean, Either[models.MemberDetails, SponsoringOrganisationDetails], SponsoringEmployerAddress, ChargeCDetails, String) => Future[Result])
+                (implicit request: DataRequest[AnyContent]): Future[Result] = {
+
+    (
+      request.userAnswers.get(isSponsoringEmployerIndividualPage),
```

Comment:
I don't think you need to retrieve isSponsoringEmployerIndividualPage since you're not doing any check on it

Reply:
Done